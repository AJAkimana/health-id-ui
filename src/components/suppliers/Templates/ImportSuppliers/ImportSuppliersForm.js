import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  Paper, Grid, Button
} from '@material-ui/core';
import UploadFileComponent from '../../../shared/UploadFileComponent';
import { ImportProductStyles } from '../../../../assets/styles/products/ImportProductStyles';
import SubmitFile from '../../../shared/SubmitFileComponent';
import FieldsTable from './FieldsTable';
import Separator from '../../../shared/Separator';


const ImportSuppliersForm = (props) => {
  const {
    state: {
      file, loading, serverResponse
    },
    onDrop,
    handleFile,
    handleUpload,
    handleDownloadTemplate,
    fileLoaded,
    isSubmitFileFailed,
    handleUploadFailed
  } = props;

  const disableDownload = !!file;

  const ButtonStyle = file ? ImportProductStyles.disabledBtn : ImportProductStyles.templateBtn;

  return (
    <Fragment>
      <Paper
        style={ImportProductStyles.paper}
      >
        <Grid
          container
          spacing={24}
          style={ImportProductStyles.importContainer}
        >
          <Grid
            item
            xs={12}
            style={ImportProductStyles.textSection}
          >
            <div>
              Import Supplier CSV
            </div>
          </Grid>
          <Grid
            item
            xs={12}
          >
            <hr style={ImportProductStyles.horizontalLine} />
          </Grid>
          <Grid
            item
            xs={12}
          >
            <div style={ImportProductStyles.templateDiv}>
              Download sample template to import suppliers
            </div>
            <div style={ImportProductStyles.templateDivBtn}>
              <Button
                style={ButtonStyle}
                disabled={disableDownload}
                onClick={handleDownloadTemplate}
              >
                DOWNLOAD SAMPLE TEMPLATE
              </Button>
            </div>
          </Grid>
          <Grid
            item
            xs={12}
          >
            <Separator />
          </Grid>
          <Grid
            item
            xs={12}
          >
            <FieldsTable />
          </Grid>
        </Grid>
      </Paper>
      {!fileLoaded ? (
        <UploadFileComponent
          onDrop={onDrop}
          handleFile={handleFile}
        />
      ) : (
        <SubmitFile
          file={file}
          loading={loading}
          handleUpload={handleUpload}
          serverResponse={serverResponse}
          isSubmitFileFailed={isSubmitFileFailed}
          handleUploadFailed={handleUploadFailed}
        />
      )
      }
    </Fragment>
  );
};

ImportSuppliersForm.propTypes = {
  state: PropTypes.instanceOf(Object).isRequired,
  fileLoaded: PropTypes.bool.isRequired,
  onDrop: PropTypes.func,
  handleFile: PropTypes.func,
  handleUpload: PropTypes.func,
  handleDownloadTemplate: PropTypes.func,
  serverResponse: PropTypes.func,
  isSubmitFileFailed: PropTypes.bool.isRequired,
  handleUploadFailed: PropTypes.func.isRequired,
};

ImportSuppliersForm.defaultProps = {
  onDrop: () => {},
  handleFile: () => {},
  handleUpload: () => {},
  handleDownloadTemplate: () => {},
  serverResponse: () => {}
};


export default ImportSuppliersForm;
