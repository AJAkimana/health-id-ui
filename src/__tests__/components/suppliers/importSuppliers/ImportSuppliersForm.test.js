import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ImportSuppliersForm from '../../../../components/suppliers/Templates/ImportSuppliers/ImportSuppliersForm';

describe('Render ImportSuppliersForm component', () => {
  const props = {
    state: {}
  };

  it('should render the ImportSuppliersForm component properly', () => {
    const setupImportSuppliersForm = () => {
      const wrapper = shallow(<ImportSuppliersForm {...props} />)
      return wrapper
    }

    let wrapper;
    wrapper = setupImportSuppliersForm();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
