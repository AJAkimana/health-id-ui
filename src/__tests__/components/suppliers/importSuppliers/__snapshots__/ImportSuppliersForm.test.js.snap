// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ImportSuppliersForm component should render the ImportSuppliersForm component properly 1`] = `
<Fragment>
  <WithStyles(Paper)
    style={
      Object {
        "marginLeft": "15%",
        "width": "70%",
      }
    }
  >
    <WithStyles(Grid)
      container={true}
      spacing={24}
      style={
        Object {
          "marginBottom": "2%",
          "marginLeft": "5%",
          "width": "90%",
        }
      }
    >
      <WithStyles(Grid)
        item={true}
        style={
          Object {
            "color": "rgba(60, 57, 57, 0.87)",
            "fontSize": "20px",
            "fontWeight": "bold",
            "marginLeft": "36%",
            "marginTop": "1%",
          }
        }
        xs={12}
      >
        <div>
          Import Supplier CSV
        </div>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <hr
          style={
            Object {
              "backgroundColor": "#E4E4E4",
              "border": "0px",
              "height": "2px",
              "marginLeft": "-3.9rem",
              "marginTop": "-0.5rem",
              "width": "114%",
            }
          }
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <div
          style={
            Object {
              "color": "#7f7d7d",
              "fontSize": "20px",
              "margin": "2% auto",
              "width": "48%",
            }
          }
        >
          Download sample template to import suppliers
        </div>
        <div
          style={
            Object {
              "margin": "0 auto",
              "marginBottom": "2%",
              "marginTop": "2%",
              "width": "27%",
            }
          }
        >
          <WithStyles(Button)
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#424242",
                "borderRadius": "15px",
                "color": "#FFFFFF",
                "marginLeft": "-1.5rem",
                "marginTop": "20px",
                "padding": "10px 25px 10px 25px",
                "width": "120%",
              }
            }
          >
            DOWNLOAD SAMPLE TEMPLATE
          </WithStyles(Button)>
        </div>
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <Separator />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        <FieldsTable />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Paper)>
  <UploadFileComponent
    handleFile={[Function]}
    onDrop={[Function]}
  />
</Fragment>
`;
